GooglePlayInstant/Editor/PlayInstantRunner.cs

```diff
@@ -0,0 +1,165 @@
+                }
+
+                return;
+            }
+
+            var failedPolicies = new List<string>(PlayInstantSettingPolicy.GetRequiredPolicies()
+                .Where(policy => !policy.IsCorrectState())
+                .Select(policy => policy.Name));
+            if (failedPolicies.Count > 0)
+            {
+                Debug.LogErrorFormat("Build and Run halted due to incompatible settings: {0}",
+                    string.Join(", ", failedPolicies.ToArray()));
+                var message = string.Format(
+                    "{0}\n\nClick \"OK\" to open the settings window and make required changes.",
+                    string.Join("\n\n", failedPolicies.ToArray()));
+                if (EditorUtility.DisplayDialog(BuildAndRunErrorTitle, message, OkButtonText, CancelButtonText))
+                {
+                    PlayerAndBuildSettingsWindow.ShowWindow();
+                }
+
+                return;
+            }
+
+            var apkPath = Path.Combine(Path.GetTempPath(), "temp.apk");
+            Debug.LogFormat("Build and Run package location: {0}", apkPath);
+
+            var buildPlayerOptions = CreateBuildPlayerOptions(apkPath);
+#if UNITY_2018_1_OR_NEWER
+            var buildReport = BuildPipeline.BuildPlayer(buildPlayerOptions);
+            switch (buildReport.summary.result)
+            {
+                case BuildResult.Cancelled:
+                    Debug.Log("Build cancelled");
+                    return;
+                case BuildResult.Succeeded:
+                    // BuildPlayer can fail and still return BuildResult.Succeeded so detect by checking totalErrors.
+                    if (buildReport.summary.totalErrors > 0)
+                    {
+                        // No need to display a message since Unity will already have done this.
+                        return;
+                    }
+
+                    // Actual success: continue on to the run step.
+                    break;
+                case BuildResult.Failed:
+                    LogError(string.Format("Build failed with {0} error(s)", buildReport.summary.totalErrors));
+                    return;
+                default:
+                    LogError("Build failed with unknown error");
+                    return;
+            }
+#else
+            var buildPlayerResult = BuildPipeline.BuildPlayer(buildPlayerOptions);
+            if (!string.IsNullOrEmpty(buildPlayerResult))
+            {
+                // Check for intended build cancellation.
+                if (buildPlayerResult == "Building Player was cancelled")
+                {
+                    Debug.Log(buildPlayerResult);
+                }
+                else
+                {
+                    LogError(buildPlayerResult);
+                }
+
+                return;
+            }
+#endif
+
+            var window = PostBuildCommandLineDialog.CreateDialog("Install and run app");
```

Comment:
Could this be block initialized or does that only work for plain-ole structs?

Reply:
I think it would only work if I was calling the constructor here. Instead I'm calling CreateDialog().